src/util/http/HttpServer.h

```diff
@@ -144,6 +144,7 @@ class HttpServer {
 
     // Wait until the posted task has successfully executed
     future.wait();
+    ioContext_.stop();
```

Comment:
Without this change sometimes the tests ran forever because all threads were asleep waiting for something to happen that never occured